client/my-sites/stats/hooks/use-stats-purchases.tsx

```diff
@@ -121,24 +123,23 @@ export default function useStatsPurchases( siteId: number | null ) {
 	const hasLoadedSitePurchases = useSelector( hasLoadedSitePurchasesFromServer );
 
 	// Determine whether a product is owned.
-	// TODO we need to do plan check as well, because Stats products would be built into other plans.
-	const isFreeOwned = useMemo( () => {
-		return isProductOwned( sitePurchases, PRODUCT_JETPACK_STATS_FREE );
-	}, [ sitePurchases ] );
+	const isFreeOwned = useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, FEATURE_STATS_FREE );
+	} );
 
 	const isCommercialOwned = useMemo(
 		() => isCommercialPurchaseOwned( sitePurchases ),
 		[ sitePurchases ]
 	);
```

Comment:
Should we rely on the `STATS_PAID` feature here from now on?

Reply:
`STATS_COMMERCIAL` seems to now be included in both products mentioned in  the `supportCommercialPurchaseUse` and `isCommercialPurchaseOwned` checks.

Does it mean effectively both `isCommercialOwned` and `supportCommercialUse` will always have the same value? Can we then remove one of these flags?